plugins/cassandra/cassandrawu.cpp

```diff
@@ -3528,11 +3530,23 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
             case WUStateDebugRunning:
             case WUStateBlocked:
             case WUStateAborting:
-                SessionId agent = getUnsignedResult(NULL, cass_row_get_column(row, 1));
-                if (agent && checkAbnormalTermination(wuid, state, agent))
+                if (agentSessionStopped)
+                {
+                    reportAbnormalTermination(wuid, state, agent);
                     return state;
+                }
+                if (queryDaliServerVersion().compare("2.1")>=0)
+                {
+                    agent = getUnsignedResult(NULL, cass_row_get_column(row, 1));
+                    if(agent && querySessionManager().sessionStopped(agent, 0))
+                    {
+                        agentSessionStopped = true;
+                        continue;
+                    }
+                }
                 break;
             }
+            agentSessionStopped = false;
```

Comment:
ah ok, makes sense, but not obvious.
Maybe a comment on this line as to why need to reset agentSessionStopped.